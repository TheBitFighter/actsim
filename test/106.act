deftype str (int<4> x)
{
   methods {
     function plus (int<2> y) : int<4>
     {
       int i;
       chp { i := 3; 
             self := x + y;
            *[ i > 0 -> self := self + i; i := i - 1 ]
           }
     }
   }
}

defproc test()
{
   str s;
   chp {
      s.x := 3;
      log ("got ", s.plus(2))
   }
}

test t;
