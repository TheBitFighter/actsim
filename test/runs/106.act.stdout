[                  10] <>  got 11
